src/pages/consultations/ConsultationDetailsPage.tsx

```diff
@@ -49,20 +74,67 @@ export const ConsultationDetailsPage = () => {
               </HStack>
             </VStack>
-            {consultation.owner.id === currentUser.id ? (
-              <Button as={Link} to={`/consultations/${consultation.id}/edit`} colorScheme="brand">
-                Szerkesztés
-              </Button>
-            ) : consultation.participants.some((p) => p.id === currentUser.id) ? (
-              <Button colorScheme="red">Mégsem megyek</Button>
-            ) : (
-              <Button colorScheme="brand">Megyek</Button>
-            )}
+
+            <VStack justify={['center', 'flex-end']} align="flex-end">
+              {(consultation.owner.id === loggedInUser!!.id ||
+                loggedInUser!!.isAdmin ||
+                consultation.presentations.some((p) => p.id === loggedInUser!!.id)) && (
+                <Button width="100%" as={Link} to={`/consultations/${consultation.id}/edit`} colorScheme="brand">
+                  Szerkesztés
+                </Button>
+              )}
+              {(consultation.owner.id === loggedInUser!!.id ||
+                loggedInUser!!.isAdmin ||
+                consultation.presentations.some((p) => p.id === loggedInUser!!.id)) && (
+                <>
+                  <ConfirmDialogButton
+                    buttonColorSchene="red"
+                    buttonText="Törlés"
+                    buttonWidth="100%"
+                    headerText="Konzultáció törlése"
+                    bodyText="Biztos törölni szeretnéd a konzultációt?"
+                    confirmButtonText="Törlés"
+                    confirmAction={deleteConsultation}
+                  />
+                </>
+              )}
+              {consultation.participants.some((p) => p.id === loggedInUser!!.id) && !consultation.presentations.some((p) => p.rating) && (
+                <>
+                  <ConfirmDialogButton
+                    buttonText="Mégsem megyek"
+                    buttonColorSchene="red"
+                    headerText="Biztos nem mész a konzira?"
+                    confirmButtonText="Nem megyek"
+                    confirmAction={() => {
+                      leaveConsultation(+consultationId!!, {
+                        onSuccess: () => {
+                          toast({ title: 'Kiléptél a konzultációból!', status: 'success' })
+                          refetch()
+                        }
+                      })
+                    }}
+                  />
+                </>
+              )}
+              {!consultation.presentations.some((p) => p.id === loggedInUser!!.id) &&
+                !consultation.participants.some((p) => p.id === loggedInUser!!.id) && (
+                  <>
+                    <Button
+                      onClick={() => {
+                        joinConsultation(+consultationId!!)
+                      }}
+                      colorScheme="brand"
+                    >
+                      Megyek
+                    </Button>
+                  </>
+                )}
```

Comment:
owner shouldn't be able to join, even if they aren't presenting. (Or maybe they should, but then change the permissions on the backend xd)